logging/lazy_fields.go

```diff
@@ -0,0 +1,188 @@
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyNode(key string, val *enode.Node) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, val.String())
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyLocalNode(key string, val *enode.LocalNode) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, val.Node().String())
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazySubnets(key string, val records.Subnets) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, val.String())
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyPeerID(key string, val peer.ID) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, val.String())
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyIP(key string, val net.IP) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, val.String())
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyMessageID(key string, val spectypes.MessageID) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, val.String())
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazySince(key string, from time.Time) zapcore.Field {
+	if showHeavyFields() {
+		return zap.Duration(key, time.Since(from))
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyCurrentSlot(key string, network beacon.Network) zapcore.Field {
+	if showHeavyFields() {
+		return zap.Uint64(key, uint64(network.EstimatedCurrentSlot()))
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyStartTime(key string, network beacon.Network, slot spec.Slot) zapcore.Field {
+	if showHeavyFields() {
+		return zap.Time(key, network.GetSlotStartTime(slot))
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyTimeFormat(key string, time time.Time, format string) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, time.Format(format))
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazyMetrics(key string, metrics *ristretto.Metrics) zapcore.Field {
+	if showHeavyFields() {
+		return zap.String(key, metrics.String())
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func LazySyncResults(key string, msgs []protocolp2p.SyncResult) zapcore.Field {
+	if showHeavyFields() {
+		var v []string
+		for _, m := range msgs {
+			var sm *specqbft.SignedMessage
+			if m.Msg.MsgType == spectypes.SSVConsensusMsgType {
+				sm = &specqbft.SignedMessage{}
+				if err := sm.Decode(m.Msg.Data); err != nil {
+					v = append(v, fmt.Sprintf("(%v)", err))
+					continue
+				}
+				v = append(
+					v,
+					fmt.Sprintf(
+						"(type=%d height=%d round=%d)",
+						m.Msg.MsgType,
+						sm.Message.Height,
+						sm.Message.Round,
+					),
+				)
+			}
+			v = append(v, fmt.Sprintf("(type=%d)", m.Msg.MsgType))
+		}
+
+		return zap.String(key, strings.Join(v, ", "))
+	}
+
+	return zap.String(key, fieldMotLogged)
+}
+
+func showHeavyFields() bool {
+	if os.Getenv("SHOW_HEAVY_FIELDS") == "true" {
+		return true
+	}
```

Comment:
if we're being very careful about performance we shouldn't be looking up this value so much. it should probably a package level const/variable